### PR TITLE
fix(core): Update spark version to 2.2.0

### PR DESCRIPTION
**Pull request by noorul**
**Pull Request checklist**

- [X] The commit(s) message(s) follows the contribution [guidelines](doc/contribution-guidelines.md#commit-message-format) ?

**Current behavior :** 

We are using spark 2.1.0

**New behavior :**

Use latest version 2.2.0

---
This change is [Reviewable](https://reviewable.io/reviews/spark-jobserver/spark-jobserver/845)

### CODE REVIEW

**Comment by velvia** [comment]
Sure.

**Comment by noorul** [comment]
Do not merge as there is some issue with test. Yet to figure out.

**Comment by noorul** [comment]
Apply wip label

**Comment by juanmirocks** [comment]
Any news on this one? The build only failed for a failed download on sbt 0.13.8 (quite old version)

**Comment by noorul** [comment]
@zeitos Python context creation is failing :(

**Comment by zeitos** [comment]
@noorul ups, I looked at code style and possible bugs is it still in WIP? (I just noticed the label)

File: job-server-python/src/python/example_jobs/hive_window/__init__.py [hunk 1]
```diff
@@ -30,3 +30,31 @@ def run_job(self, context, runtime, data):
...
+class HiveWindowSessionJob(SparkJob):
+
+    def validate(self, session, runtime, config):
+        problems = []
+        job_data = None
+        if config.get('input.data', None):
+            job_data = config.get('input.data')
+        else:
+            problems.append('config input.data not found')
+        if len(problems) == 0:
```
Comment (zeitos) [inline job-server-python/src/python/example_jobs/hive_window/__init__.py]: I think  
```
if not problems
```
is more readable

Comment (zeitos) [reply job-server-python/src/python/example_jobs/hive_window/__init__.py]: and you usually do:
```
return job_data if not problems else  build_problems(problems)
```

File: job-server-python/src/python/example_jobs/hive_window/__init__.py [hunk 2]
```diff
@@ -30,3 +30,31 @@ def run_job(self, context, runtime, data):
...
         return [(r[0], r[1], r[2]) for r in results]
+
+
+class HiveWindowSessionJob(SparkJob):
+
+    def validate(self, session, runtime, config):
+        problems = []
+        job_data = None
+        if config.get('input.data', None):
+            job_data = config.get('input.data')
```
Comment (zeitos) [inline job-server-python/src/python/example_jobs/hive_window/__init__.py]: you can just initialize job_data as the previous get
``` 
job_data = config.get('input.data', None)
if not job_data:
    problems.append
```

**Comment by noorul** [comment]
@zeitos I removed wip label. Can you take a look again?

**Comment by noorul** [comment]
@zeitos I removed that new class. Merging this!

**Comment by noorul** [comment]
@juanmirocks Merged!

**Comment by juanmirocks** [comment]
@noorul & @all you rock!!

**Comment by hyitclj** [comment]
why I don's use it ? detail in https://github.com/spark-jobserver/spark-jobserver/issues/936

**Pull request merged**